sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -33,7 +34,7 @@ import org.apache.spark.sql.catalyst.catalog.{CatalogRelation, CatalogTable}
  * Right now, it only supports Hive tables and it only updates the size of a Hive table
  * in the Hive metastore.
  */
-case class AnalyzeTableCommand(tableName: String) extends RunnableCommand {
+case class AnalyzeTableCommand(tableName: String, noscan: Boolean = true) extends RunnableCommand {
```

Comment:
Collecting statistics has a cost, so you would normally do this every once in a while, and not on every change to your data. The underlying idea is that the distribution of data typically doesn't change to much.

If we were to store the input statistics aggregation buffers somewhere, you could actually update statistics easily after an insert (you only need to scan the inserted data).